src/UI/Component/Legacy/Legacy.php

```diff
@@ -16,4 +18,21 @@ interface Legacy extends \ILIAS\UI\Component\Component
      * @return	string
      */
     public function getContent();
+
+    /**
+     * Get a legacy component like this, but with an additional signal with custom JavaScript code
+     *
+     * @param $signal_name
+     * @param $js_code
+     * @return Legacy
+     */
+    public function withCustomSignal(string $signal_name, string $js_code) : Legacy;
+
+    /**
+     * Get signal with custom JavaScript code
+     *
+     * @param $signal_name
+     * @return Signal|null
```

Comment:
I'd rather throw an Exception than silently return null.

Reply:
I added now a throws \InvalidArgumentException. This was the best fitting one I had in mind.